core/WFCORE-5343_Support_YAML_files_for_configuration_customization.adoc

```diff
@@ -0,0 +1,255 @@
+Sample description of a correct deployment:
+----
+wildfly-configuration:
+    deployment:
+        test.jar:
+            content:
+                - 
+                    path: test.jar
+                    relative-to: jboss.server.base.dir
+                    archive: true
+        hello.jar:
+            content:
+                - 
+                    path: test.jar
+                    relative-to: jboss.server.base.dir
+                    archive: true
+----
+
+But such a description must fail:
+----
+wildfly-configuration:
+    deployment:
+        test.jar:
+            content:
+                - 
+                    path: test.jar
+                    relative-to: jboss.server.base.dir
+                    archive: true
+        hello.jar:
+            content:
+                -
+                    empty: true
+----
+
+Note that any failure will cause the boot to fail, thus this description will fail with an IllegalArgumentException describing the failure.
+
+== Test Plan
+
+org.jboss.as.test.manualmode.management.persistence.yaml.YamlExtensionTestCase in the manual test suite holds most of the testing:
+
+ * YamlExtensionTestCase#testNoYaml: `-y`, no yaml file(s) defined
+ * YamlExtensionTestCase#testSimpleYaml: checking thata single yaml file is properly applied.
+ * YamlExtensionTestCase#testAttributeOverrideByTwoYamlFiles: checking that multiple yaml files are properly applied.
+ * YamlExtensionTestCase#testSimpleYaml#testAddingResourceWithOverridingAttributesByTwoYamlFiles: checking that removing then adding a resource works.
+ * YamlExtensionTestCase#testYamlOperations: test the proper behaviour of the tags `!undefine`, `!remove` and `!list-add`).
+ * YamlExtensionTestCase#testUndefineNonExistentAttributeYamlOperations checks that you can't undefined something that doesn't exist.
+ * YamlExtensionTestCase#testListAddOperationToStringFails: test that adding a wrong type element fails.
+ * YamlExtensionTestCase#testListAddOperationToNonExistentResourceFails: test that adding an elemnt to a unexisting attribute fails.
+ * YamlExtensionTestCase#testReplacingResourceByEmptyResourceLogsWarning: test tha adding an empty resource via YAML logs a warning.
+ * YamlExtensionTestCase#testSimpleYamlWithReload checks that the YAML changes are not persisted.
+ * YamlExtensionTestCase#testSimpleYamlWithCliBootOps: checks that YAML and cli scripts are working properly together.
+ * YamlExtensionTestCase#testSimpleYaml#testDeploymentYaml: checks that adding unmanaged deployments via YAML works.
+ * YamlExtensionTestCasetestServerStartFailedForManagedDeployment: checks that managed deployments can't be added via YAML.
+ * YamlExtensionTestCase#testAddingExtensionPathDeploymentOverlayYamlLogsWarnings: checks that path, extension and deployment-overlay can't be added via YAML.
+
+
+== Community Documentation
+
+https://www.wildfly.org/news/2022/04/26/YAML-configuration-extension/
```

Comment:
@ehsavoie I would say this should be https://github.com/wildfly/wildfly/pull/17970. having the article linked is good as well